Tests/Service/Request/RequestFactoryTest.php

```diff
@@ -166,12 +166,14 @@ public function testBuildFlow()
     public function testBuildFlowWithQueryParams()
     {
         $baseUrl = 'baseUrl';
-        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value';
+        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value&arr-param[key]={arrParam}&dot.seprate[0].array={dotSeparatedArrayParam}';
```

Comment:
Could we update the test for PHP way of representing nested arrays in the URL?
Like `&arr-param[key1]=arrValue1&arr-param[key2]=arrayValue2` 

Since all these `[0]` elements and "dot separation quite tricky.

Reply:
Updated.